### PR TITLE
Add catch for invalid spec URLs

### PR DESCRIPTION
**Pull request by jamesnw**
Previously, a missing or invalid spec URL would throw an error without a message. This change surfaces that error when running `npm run test:specs`.

In addition, the test was reporting the number of specs as the number of features, so now it reports both.

### CODE REVIEW

File: scripts/specs.ts [hunk 1]
```diff
@@ -92,19 +93,25 @@ for (const [allowedUrl, message] of defaultAllowlist) {
 for (const [id, data] of Object.entries(features)) {
     const specs = Array.isArray(data.spec) ? data.spec : [data.spec];
     for (const spec of specs) {
-        const url = new URL(spec);
-        if (!isOK(url)) {
+        try {
+            var url = new URL(spec);
```
Comment (ddbeck) [inline scripts/specs.ts:97]:
```suggestion
            const url = new URL(spec);
```

**Review by ddbeck — DISMISSED** [review]
Oh excellent, thank you!

File: scripts/specs.ts (hunk 1 above)
Comment (ddbeck) [reply scripts/specs.ts:97]: Ha, so I started to say something about this and forgot about it.

I think `var` is pretty confusing—it's weird that it's possible to reference it outside the scope here. Would you mind changing this a `let url` (where the original `const url = URL(spec);` was), then reassigning on this line?

```suggestion
            url = new URL(spec);
```

**Review by ddbeck — CHANGES_REQUESTED** [review]

File: scripts/specs.ts (hunk 1 above)
Comment (jamesnw) [reply scripts/specs.ts]: Makes sense to me! Updated.

**Review by ddbeck — APPROVED** [review]
Thank you!

**Pull request merged**